lessons/AI/ai-linear-regression/langs/en-us/index.adoc

```diff
@@ -0,0 +1,223 @@
+- Training images were reduced in resolution to 30x32 pixels and provided as inputs associated with the corresponding "correct" steering angles as their outputs.
+
+The human driver in this context is often referred to as the _supervisor_; it is their tagging of input-output pairs that makes supervised machine learning possible.
+
+Once ALVINN's training data has been collected, it can be represented with a table, as below.
+
+@teacher{
+Note: These images were selected for illustrative purposes. The 30x32 pixel images produced by ALVINN were of much lower quality!
+}
+
+[cols="^.^1,^.^1", stripes="none", options="header"]
+|===
+
+| x (input data) | y (output data)
+| @image{images/road1.png, 100} 	| 0.02º
+| @image{images/road2.png, 100} 	| -95.2º
+| @image{images/road3.png, 100} 	| 135.43º
+
+|===
+
+
+ALVINN then uses the set of inputs and outputs to create a @vocab{function}, which we can summarize with contract notation:
+
+@show{(contract 'steer '((road-image Image)) "Number")}
+
+@QandA{
+@Q{What information does ALVINN process before producing a steering angle?}
+@A{ALVINN instantaneously processes the image of the road in front of him in order to produce a steering angle.}
+
+@Q{When humans make decisions about which way to steer, we are working with a lot of other information besides the image of the road in the moment. What examples can you come up with?}
+
+@A{We can use maps to give us a sense of what the road might do before we can see it.}
+@A{If a road is curving, we can predict that it will continue to curve.}
+@A{We know that roads don't curve beyond a certain degree--and if they do, we can expect to see a warning sign ahead of time.}
+@A{We can look at the cars in front of us to better understand how we should steer.}
+@A{If we know that we're going to be taking a left turn, we might switch lanes and slow down ahead of time.}
+}
+
+@lesson-point{
+ALVINN's function dramatically reduces the complexity of the data and predicts the value of the @vocab{response variable} as though it is completely dependent on the @vocab{explanatory variable}.
+}
+
+After ALVINN has done some reasonable amount of training, a human pushes ALVINN's run switch and the vehicle begins driving.
+
+As ALVINN drives, it produces steering angles it learned from the human driver (who demonstrated the steering angle at each segment of the road), along with a numeric measure of "confidence" in the angle provided.
+
+=== Investigate
+
+The process described above is an example of @vocab{regression}, a machine learning algorithm that can be trained to predict continuous real-number outputs (like steering angle!).
+
+@vocab{Linear regression}, which you may already have heard of or worked with, is one of the most basic types of regression, because it involves the use of a **_linear_** @vocab{predictor function}. Machine learning incorporates other more complex nonlinear regression types, too (for example: quadratic, polynomial, etc).
+
```

Comment:
Perhaps my understanding of linear regression is limited, but I feel like this brief discussion of regression could really use some acknowledgement of the fact that lr focuses on numerical data and machine learning is not only powerful enough to work with curves, but it can also work with other data types, such as images.